File: python/app/routers/health.py
from datetime import datetime, timezone
from fastapi import APIRouter

router = APIRouter()


@router.get("/health")
def health():
    return {"status": "ok", "service": "naves-python", "ts": datetime.now(timezone.utc).isoformat()}


@router.get("/health/deep")
def health_deep():
    from ..db import get_supabase_admin
    checks = {"service": "ok"}
    try:
        sb = get_supabase_admin()
        res = sb.table("cohortes").select("id").limit(1).execute()
        checks["supabase"] = {"ok": True, "rows_seen": len(res.data or [])}
    except Exception as e:
        checks["supabase"] = {"ok": False, "error": str(e)}
    return {"status": "ok", "checks": checks}

File: backend/src/routes/health.ts
import { Router } from 'express';
import { supabaseAdmin } from '../db/supabase.js';

const router = Router();

router.get('/health', (_req, res) => {
  res.json({ status: 'ok', service: 'naves-backend', ts: new Date().toISOString() });
});

router.get('/health/deep', async (_req, res) => {
  const checks: Record<string, unknown> = { service: 'ok' };
  try {
    const { error } = await supabaseAdmin.from('cohortes').select('id').limit(1);
    checks.supabase = error ? { ok: false, error: error.message } : { ok: true };
  } catch (e) {
    checks.supabase = { ok: false, error: (e as Error).message };
  }
  res.json({ status: 'ok', checks, ts: new Date().toISOString() });
});

export default router;
